### PR TITLE
Implement pytest plugin to dump Apiritif traces

### PR DESCRIPTION
**Pull request by undera**

### CODE REVIEW

File: setup.py [hunk 1]
```diff
@@ -31,5 +31,11 @@
     download_url='https://github.com/Blazemeter/apiritif',
     docs_url='https://github.com/Blazemeter/apiritif',
 
-    install_requires=['nose', 'requests>=2.11.1', 'jsonpath-rw', 'lxml', 'unicodecsv'],
+    install_requires=['nose', 'pytest', 'requests>=2.11.1', 'jsonpath-rw', 'lxml', 'unicodecsv'],
+
+    entry_points={
+        'pytest11': [
+            'pytest_apiritif = apiritif.pytest_plugin',
```
[the remark this answers is not in the document]
Comment (undera) [reply setup.py]: This is PyTest plugin loading interface via setuptools entry points. It is official way to add plugins: https://docs.pytest.org/en/latest/writing_plugins.html#plugin-discovery-order-at-tool-startup

**Comment by undera** [comment]
The structure of Apiritif trace is complex, it's a series of Samples with subsamples. I call this sequence as "trace" because it is really a reflection of Apiritif events that happened.
The usecase is to deeper analysis of test results, to figure out which step of multi-step test has failed and for which reason.

File: apiritif/pytest_plugin.py [hunk 2]
```diff
@@ -0,0 +1,79 @@
...
+
+
+def pytest_unconfigure(config):
+    name = ApiritifPlugin.__name__
+    if config.pluginmanager.has_plugin(name):
+        plugin = config.pluginmanager.get_plugin(name)
+        config.pluginmanager.unregister(plugin)
+
+
+class ApiritifPlugin(object):
```
[the remark this answers is not in the document]
Comment (undera) [reply apiritif/pytest_plugin.py]: The class has been renamed, please review

**Comment by greyfenrir** [comment]
thanks for explanations, merged

**Pull request merged**